programs/nft-club/src/components/creator.rs

```diff
@@ -0,0 +1,99 @@
+use anchor_lang::prelude::*;
+
+// 
+// Endpoints
+// 
+
+/*
+ * Create account and benefit in same function
+ * creator specifies the actual benefits in form when sign up
+ *
+ * Can program instructions handle variable amount of benefits
+ * Benefit seeds are passed in here as parameters
+ */
+/* program.rpc.createAccount on frontend*/
+pub fn create_account(ctx: Context<CreateAccount>, username: String, email: String, description: String, num_benefits: u8) -> ProgramResult {
+    let creator: &mut Account<Creator> = &mut ctx.accounts.creator;
+    let authority: &Signer = &ctx.accounts.authority;
+
+    if username.chars().count() > 42 {
+        return Err(ErrorCode::UsernameTooLong.into())
+    }
+
+    if email.chars().count() > 42 {
+        return Err(ErrorCode::EmailTooLong.into())
+    }
+
+    if description.chars().count() > 420 {
+        return Err(ErrorCode::DescriptionTooLong.into())
+    }
+
+    creator.authority = *authority.key;
+    creator.username = username;
+    creator.email = email;
+    creator.description = description;
+    creator.num_benefits = num_benefits;
+
+    Ok(())
+}
+
+// 
+// Data Validators
+// 
+#[derive(Accounts)]
+pub struct CreateAccount<'info> {
+    // Create account of type Creator and assign creators's pubkey as the payer
+    #[account(init, payer = authority, space = Creator::LEN)]
+    pub creator: Account<'info, Creator>,
+
+    // Define user as mutable - money in their account, profile data, etc.
+    #[account(mut)]
+    pub authority: Signer<'info>,
+
+    // Ensure System Program is the official one from Solana.
+    pub system_program: Program<'info, System>,
+}
+
+// 
+// Data Structures
+// 
+#[account]
+pub struct Creator {
+    pub authority: Pubkey,
+    pub username: String,
+    pub email: String,
```

Comment:
Just want to point out that here you're asking people to associate their email with their wallet address... Some privacy conscious users may not like that. That's a design choice we'll have to consider, pros/cons of adding emails